scipy/ndimage/_measurements.py

```diff
@@ -307,6 +307,91 @@ def find_objects(input, max_label=0):
     return _nd_image.find_objects(input, max_label)
 
 
+def value_indices(arr, *, nullval=None):
+    """
+    Find indices of each distinct value in given array.
+
+    Parameters
+    ----------
+    arr : ndarray of ints
+        Array containing integer values.
```

Comment:
Are we talking about n-D arrays? or just 2D?

If n-D, then we need more tests. At least: 1D, 2D, 3 or more D.

Reply:
Yes, definitely a general n-D array input. Good point, I will dream up some more test cases.